### PR TITLE
add hostname prefix for appgw backends

### PR DESCRIPTION
**Pull request by adusumillipraveen**
**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by timja — COMMENTED** [review]
why?

**Comment by adusumillipraveen** [comment]
> why?

trying to add docmosis.platform.hmcts.net (which isn't supported in current state)

**Comment by timja** [comment]
> > why?
> 
> trying to add docmosis.platform.hmcts.net (which isn't supported in current state)

put it under a product? dg or em maybe?

seems like quite a lot of complexity added here, or just add a custom domain name override?

**Comment by adusumillipraveen** [comment]
> > > why?
> > 
> > 
> > trying to add docmosis.platform.hmcts.net (which isn't supported in current state)
> 
> put it under a product? dg or em maybe?
> 
> seems like quite a lot of complexity added here, or just add a custom domain name override?

even it didn't like it tbh, i was trying to set a full_name before it's passed to the for_each blocks, but wasnt easy to do that.

Anyway, how does the latest change look ?

File: modules/app-gateway-module-backend/main.tf [hunk 1]
```diff
@@ -77,8 +77,8 @@ resource "azurerm_application_gateway" "ag" {
     for_each = [for app in local.gateways[count.index].app_configuration : {
       name          = "${app.product}-${app.component}"
       path          = lookup(app, "health_path_override", "/health/liveness")
-      host_name     = "${app.product}-${app.component}-${var.env}.${local.gateways[count.index].gateway_configuration.host_name_suffix}"
-      ssl_host_name = "${app.product}-${app.component}.${local.gateways[count.index].gateway_configuration.ssl_host_name_suffix}"
+      host_name     = "lookup(app, 'host_name_prefix', ${app.product}-${app.component}-${var.env}).${local.gateways[count.index].gateway_configuration.host_name_suffix}"
```
Comment (timja) [inline modules/app-gateway-module-backend/main.tf:80]: isn't this a host_name_override? not a prefix?

Comment (adusumillipraveen) [reply modules/app-gateway-module-backend/main.tf:80]: hmm, used it as we used host_name_suffix here
 https://github.com/hmcts/azure-platform-terraform/blob/master/environments/sbox/backend_lb_config.yaml#L5 . and we still use it as a suffix after this. May be that should be domain_name?

Comment (timja) [reply modules/app-gateway-module-backend/main.tf:80]: where are you using it as a suffix? not sure if i'm missing something but it looks like `host_name_prefix` is the whole domain name

**Review by timja — APPROVED** [review]

File: modules/app-gateway-module-backend/main.tf (hunk 1 above)
Comment (timja) [reply modules/app-gateway-module-backend/main.tf:80]: i missed the )

**Review by thomast1906 — APPROVED** [review]

**Pull request merged**